pkg/plugins/manifest.go

```diff
@@ -100,20 +103,24 @@ func GetPluginSignatureState(plugin *PluginBase) PluginSignature {
 	}
 
 	// Verify the manifest contents
+	log.Debug("Verifying contents of plugin manifest...", "plugin", plugin.Id)
 	for p, hash := range manifest.Files {
 		// Open the file
-		f, err := os.Open(path.Join(plugin.PluginDir, p))
+		fp := path.Join(plugin.PluginDir, p)
+		f, err := os.Open(fp)
 		if err != nil {
 			return PluginSignatureModified
 		}
 		defer f.Close()
 
 		h := sha256.New()
 		if _, err := io.Copy(h, f); err != nil {
+			log.Warn("Couldn't read plugin file", "plugin", plugin.Id, "filename", fp)
 			return PluginSignatureModified
 		}
-		sum := string(h.Sum(nil))
+		sum := hex.EncodeToString(h.Sum(nil))
```

Comment:
How did this work before? :/

Reply:
@bergquist I think it was simply broken? When I updated a test's plugin to have a signed manifest, it got so far as to hit this bug.